internal_tests/pytests/grid_stat/test_grid_stat_wrapper.py

```diff
@@ -51,6 +51,46 @@ def set_minimum_config_settings(config):
     config.set('config', 'OBS_VAR1_NAME', obs_name)
     config.set('config', 'OBS_VAR1_LEVELS', obs_level)
 
+@pytest.mark.parametrize(
```

Comment:
This logic is interesting and caught my eye. I'm assuming it's critical to how the various scenarios are evaluated of fcst/obs probablity and the various tools now having their own settings. 
Can you summarize the logic of this for me? For example, if "FCST_IS_PROB" is False but "FCST_GRID_STAT_IS_PROB" is True, how is that handled? And perhaps more importantly, how is the reverse handled: where FCST_GRID_STAT_IS_PROB is False but FCST_IS_PROB is True? What about the scenario where GridStat is the only tool called but these conditions are present?

Reply:
These are the different values that are passed to the unit test. In this test, there is a list of tuples that contain config overrides and expected values. For each tuple in the list, the values are passed to the test function and used in that call to the test. The values in config_overrides are set in the METplusConfig before initializing the wrapper. The values in expected_values are compared to the values set in the wrapper's c_dict dictionary to make sure they are set as expected.

The expected logic is:
* If the wrapper specific version (FCST_GRID_STAT_IS_PROB) is set, use that value. Otherwise:
* If the generic version (FCST_IS_PROB) is set, use that value. Otherwise:
* Set the variable to False

The first example you provided ("FCST_IS_PROB" is False but "FCST_GRID_STAT_IS_PROB" is True) is not explicitly tested here, so it wouldn't hurt to add it. I can do that. Test 5 ensures that the wrapper-specific version is read properly when the generic is unset ("FCST_IS_PROB" is unset but "FCST_GRID_STAT_IS_PROB" is True).

Test 3 covers your 2nd example: FCST_GRID_STAT_IS_PROB is False but FCST_IS_PROB is True

If GridStat is the only tool called, then the same rules listed above apply.

Reply:
Excellent, thank you for the explanation and additional test.